### ISSUE TITLE
Communication between cvat and cvat_db timeout when superuser being created

### ISSUE THREAD

**Issue by waterfall414** [openvinotoolkit/cvat#3179]
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

I am trying to deploy cvat behind corporate firewall for internal network usage. I pulled the latest develop branch, followed the installation guide and got stuck when trying to create superuser. The error seems to be related to network time-out. 

### Expected Behaviour

```
docker exec -it cvat bash -ic 'python3 ~/manage.py createsuperuser'
```
Then successfully created superuser

### Current Behaviour

Error after three-minute waiting 
```
> docker exec -it cvat bash -ic 'python3 ~/manage.py createsuperuser'
INFO - 2021-05-11 21:44:49,630 - font_manager - generated new fontManager
System check identified some issues:

WARNINGS:
?: (urls.W005) URL namespace 'v1' isn't unique. You may not be able to reverse all URLs in this namespace
Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/django/db/backends/base/base.py", line 219, in ensure_connection
    self.connect()
  File "/opt/venv/lib/python3.8/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.8/site-packages/django/db/backends/base/base.py", line 200, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/opt/venv/lib/python3.8/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.8/site-packages/django/db/backends/postgresql/base.py", line 187, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/opt/venv/lib/python3.8/site-packages/psycopg2/__init__.py", line 127, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: could not connect to server: Connection timed out
	Is the server running on host "cvat_db" (172.28.0.3) and accepting
	TCP/IP connections on port 5432?


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/django/manage.py", line 21, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/opt/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/venv/lib/python3.8/site-packages/django/core/management/base.py", line 330, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/venv/lib/python3.8/site-packages/django/contrib/auth/management/commands/createsuperuser.py", line 79, in execute
    return super().execute(*args, **options)
  File "/opt/venv/lib/python3.8/site-packages/django/core/management/base.py", line 370, in execute
    self.check_migrations()
  File "/opt/venv/lib/python3.8/site-packages/django/core/management/base.py", line 459, in check_migrations
    executor = MigrationExecutor(connections[DEFAULT_DB_ALIAS])
  File "/opt/venv/lib/python3.8/site-packages/django/db/migrations/executor.py", line 18, in __init__
    self.loader = MigrationLoader(self.connection)
  File "/opt/venv/lib/python3.8/site-packages/django/db/migrations/loader.py", line 53, in __init__
    self.build_graph()
  File "/opt/venv/lib/python3.8/site-packages/django/db/migrations/loader.py", line 216, in build_graph
    self.applied_migrations = recorder.applied_migrations()
  File "/opt/venv/lib/python3.8/site-packages/django/db/migrations/recorder.py", line 77, in applied_migrations
    if self.has_table():
  File "/opt/venv/lib/python3.8/site-packages/django/db/migrations/recorder.py", line 55, in has_table
    with self.connection.cursor() as cursor:
  File "/opt/venv/lib/python3.8/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.8/site-packages/django/db/backends/base/base.py", line 259, in cursor
    return self._cursor()
  File "/opt/venv/lib/python3.8/site-packages/django/db/backends/base/base.py", line 235, in _cursor
    self.ensure_connection()
  File "/opt/venv/lib/python3.8/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.8/site-packages/django/db/backends/base/base.py", line 219, in ensure_connection
    self.connect()
  File "/opt/venv/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/opt/venv/lib/python3.8/site-packages/django/db/backends/base/base.py", line 219, in ensure_connection
    self.connect()
  File "/opt/venv/lib/python3.8/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.8/site-packages/django/db/backends/base/base.py", line 200, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/opt/venv/lib/python3.8/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.8/site-packages/django/db/backends/postgresql/base.py", line 187, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/opt/venv/lib/python3.8/site-packages/psycopg2/__init__.py", line 127, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
django.db.utils.OperationalError: could not connect to server: Connection timed out
	Is the server running on host "cvat_db" (172.28.0.3) and accepting
	TCP/IP connections on port 5432?
```

### Possible Solution

It may due to the firewall/network setting in the corporate server. I can build and run cvat in my home linux.
Additionally, I tried:
1) ping cvat_db(172.28.0.3) from bash, and it failed
2) ping cvat(172.28.0.4) from cvat_db(172.28.0.3) container, and it also failed.
3) created ~/.docker/config.json and add proxy as below. Not helpful
```json
{
  "proxies": {
    "default": {
      "httpProxy": "http://xxx:xxx",
      "httpsProxy": "https://xxx:xxx",
      "noProxy": "172.*"
    }
  }
}
```
### Context

This issue occurred when I am trying to deploy cvat in corporate research setting. So I guess other potential users may also run into a similar problem.

### Your Environment

- Git hash commit (`git log -1`): commit 7a4980619ccb1c8896ba1a0b948225ffd3864195
- Docker version `docker version` (e.g. Docker 17.0.05): 20.10.2
- Are you using Docker Swarm or Kubernetes? No
- Operating System and version (e.g. Linux, Windows, MacOS): Ubuntu 20.04
- Code example or link to GitHub repo or gist to reproduce problem: N/A
- Other diagnostic information / logs:
    <details>
    <summary>Logs from `cvat` container</summary>
  2021-05-11 21:28:04,765 INFO RPC interface 'supervisor' initialized
  2021-05-11 21:28:04,765 CRIT Server 'unix_http_server' running without any HTTP authentication checking
  2021-05-11 21:28:04,766 INFO supervisord started with pid 1
  2021-05-11 21:28:05,769 INFO spawned: 'ssh-agent' with pid 8
  2021-05-11 21:28:05,771 INFO spawned: 'clamav_update' with pid 9
  2021-05-11 21:28:05,773 INFO spawned: 'git_status_updater' with pid 10
  2021-05-11 21:28:05,775 INFO spawned: 'rqscheduler' with pid 11
  2021-05-11 21:28:05,777 INFO spawned: 'rqworker_default_0' with pid 12
  2021-05-11 21:28:05,779 INFO spawned: 'rqworker_default_1' with pid 14
  2021-05-11 21:28:05,784 INFO spawned: 'rqworker_low' with pid 17
  2021-05-11 21:28:05,786 INFO spawned: 'runserver' with pid 19
  2021-05-11 21:28:05,787 DEBG fd 10 closed, stopped monitoring <POutputDispatcher at 139978297163056 for <Subprocess at 139978297162384 with name clamav_update in state STARTING> (stdout)>
  2021-05-11 21:28:05,787 DEBG fd 14 closed, stopped monitoring <POutputDispatcher at 139978296924432 for <Subprocess at 139978297162384 with name clamav_update in state STARTING> (stderr)>
  2021-05-11 21:28:05,787 INFO exited: clamav_update (exit status 0; not expected)
  2021-05-11 21:28:05,789 DEBG received SIGCHLD indicating a child quit
  2021-05-11 21:28:05,789 DEBG 'ssh-agent' stdout output:
  SSH_AUTH_SOCK=/tmp/ssh-agent.sock; export SSH_AUTH_SOCK;
  echo Agent pid 8;
  
  2021-05-11 21:28:05,790 DEBG 'ssh-agent' stderr output:
  debug2: fd 3 setting O_NONBLOCK
  
  2021-05-11 21:28:05,792 DEBG 'rqscheduler' stderr output:
  wait-for-it.sh: waiting for cvat_redis:6379 without a timeout
  
  2021-05-11 21:28:05,794 DEBG 'rqworker_default_0' stderr output:
  wait-for-it.sh: waiting for cvat_redis:6379 without a timeout
  
  2021-05-11 21:28:05,796 DEBG 'git_status_updater' stderr output:
  wait-for-it.sh: waiting for cvat_redis:6379 without a timeout
  
  2021-05-11 21:28:05,798 DEBG 'rqworker_default_1' stderr output:
  wait-for-it.sh: waiting for cvat_redis:6379 without a timeout
  
  2021-05-11 21:28:05,800 DEBG 'rqworker_low' stderr output:
  wait-for-it.sh: waiting for cvat_redis:6379 without a timeout
  
  2021-05-11 21:28:05,801 DEBG 'runserver' stderr output:
  wait-for-it.sh: waiting for cvat_db:5432 without a timeout
  
  2021-05-11 21:28:06,802 INFO success: ssh-agent entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
  2021-05-11 21:28:06,804 INFO spawned: 'clamav_update' with pid 47
  2021-05-11 21:28:06,805 INFO success: git_status_updater entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
  2021-05-11 21:28:06,805 INFO success: rqscheduler entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
  2021-05-11 21:28:06,805 INFO success: rqworker_default_0 entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
  2021-05-11 21:28:06,805 INFO success: rqworker_default_1 entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
  2021-05-11 21:28:06,805 INFO success: rqworker_low entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
  2021-05-11 21:28:06,805 INFO success: runserver entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
  2021-05-11 21:28:06,813 DEBG fd 10 closed, stopped monitoring <POutputDispatcher at 139978296924576 for <Subprocess at 139978297162384 with name clamav_update in state STARTING> (stdout)>
  2021-05-11 21:28:06,813 DEBG fd 16 closed, stopped monitoring <POutputDispatcher at 139978296924384 for <Subprocess at 139978297162384 with name clamav_update in state STARTING> (stderr)>
  2021-05-11 21:28:06,813 INFO exited: clamav_update (exit status 0; not expected)
  2021-05-11 21:28:06,814 DEBG received SIGCHLD indicating a child quit
  2021-05-11 21:28:08,818 INFO spawned: 'clamav_update' with pid 48
  2021-05-11 21:28:08,827 DEBG fd 10 closed, stopped monitoring <POutputDispatcher at 139978297163056 for <Subprocess at 139978297162384 with name clamav_update in state STARTING> (stdout)>
  2021-05-11 21:28:08,827 DEBG fd 16 closed, stopped monitoring <POutputDispatcher at 139978296924528 for <Subprocess at 139978297162384 with name clamav_update in state STARTING> (stderr)>
  2021-05-11 21:28:08,827 INFO exited: clamav_update (exit status 0; not expected)
  2021-05-11 21:28:08,827 DEBG received SIGCHLD indicating a child quit
  2021-05-11 21:28:11,834 INFO spawned: 'clamav_update' with pid 49
  2021-05-11 21:28:11,845 DEBG fd 10 closed, stopped monitoring <POutputDispatcher at 139978296784544 for <Subprocess at 139978297162384 with name clamav_update in state STARTING> (stdout)>
  2021-05-11 21:28:11,846 DEBG fd 16 closed, stopped monitoring <POutputDispatcher at 139978296924624 for <Subprocess at 139978297162384 with name clamav_update in state STARTING> (stderr)>
  2021-05-11 21:28:11,846 INFO exited: clamav_update (exit status 0; not expected)
  2021-05-11 21:28:11,846 DEBG received SIGCHLD indicating a child quit
  2021-05-11 21:28:12,847 INFO gave up: clamav_update entered FATAL state, too many start retries too quickly

    </details>

### Next steps
You may [join our Gitter](https://gitter.im/opencv-cvat/public) channel for community support.

**Comment by nmanovic** [comment]
@waterfall414, thanks for the feedback. Let's investigate the issue together. It looks like a problem with configuration inside a complex enterprise environment. We also have proxy inside Intel and CVAT works properly.

Could you please check that all arguments are set properly? It is slightly modified output of our internal instance.

```sh
docker inspect cvat | grep -i proxy
                "no_proxy=nuclio,logstash,intel.com,.intel.com,localhost,127.0.0.1",
                "http_proxy=http://xxx.intel.com:911",
                "https_proxy=http://xxx.intel.com:911",
                "socks_proxy=",
```

Please read here as well: https://github.com/openvinotoolkit/cvat/blob/df175a856179f1c31ac2b15c80d98986a3d35acf/cvat/apps/documentation/installation.md#deploying-cvat-behind-a-proxy

**Comment by nmanovic** [comment]
@azhavoro , please look at the issue as well. Need to help.

**Comment by azhavoro** [comment]
I think the cause is the proxy settings, please note that `noProxy: “172. *"` does not work as you expect and this may be the reason ping to an IP address is not working,` no_proxy` should be a comma separated list of hostnames or IP addresses, try setting it like Nikita suggested, but add also `cvat_db` hostname

**Comment by azhavoro** [comment]
@waterfall414 Also please attach logs from `cvat_db` and `cvat` containers: `docker logs cvat > cvat.log`, `docker logs cvat_db > cvat.db.log`

**Comment by waterfall414** [comment]
Thanks @nmanovic @azhavoro for replying! 

Setting the no_proxy to "172.*" is an attempt (failed) after the problem occurred. I tried to follow the [deploying-cvat-behind-a-proxy]( https://github.com/openvinotoolkit/cvat/blob/df175a856179f1c31ac2b15c80d98986a3d35acf/cvat/apps/documentation/installation.md#deploying-cvat-behind-a-proxy) and see if it can solve my problem. In other words, the communication failed when the proxy for docker containers was not set in ~/.docker/config.json.

Before I set the proxy
```
> docker inspect cvat | grep -i proxy
                "http_proxy=",
                "https_proxy=",
                "no_proxy=nuclio,",
                "socks_proxy=",
> docker inspect cvat_db | grep -i proxy
>
```

After I set the proxy with config.json (corporate-specific information is replaced with "aaa")
```json
{
  "proxies": {
    "default": {
      "httpProxy": "http://aaa.com:9480",
      "httpsProxy": "http://aaa.com:9480"
    }
  }
}
```
```
> docker inspect cvat | grep -i proxy
                "http_proxy=http://aaa.com:9480",
                "HTTP_PROXY=http://aaa.com:9480",
                "https_proxy=http://aaa.com:9480",
                "HTTPS_PROXY=http://aaa.com:9480",
                "HTTP_PROXY=http://aaa.com:9480",
                "http_proxy=http://aaa.com:9480",
                "HTTPS_PROXY=http://aaa.com:9480",
                "https_proxy=http://aaa.com:9480",
                "no_proxy=nuclio,",
                "socks_proxy=",
> docker inspect cvat_db | grep -i proxy
                "http_proxy=http://aaa.com:9480",
                "HTTP_PROXY=http://aaa.com:9480",
                "https_proxy=http://aaa.com:9480",
                "HTTPS_PROXY=http://aaa.com:9480",
                "HTTP_PROXY=http://aaa.com:9480",
                "http_proxy=http://aaa.com:9480",
                "HTTPS_PROXY=http://aaa.com:9480",
                "https_proxy=http://aaa.com:9480",
```
I ran the commands @azhavoro suggested to dump the logs. Here they are
[cvat.log](https://github.com/openvinotoolkit/cvat/files/6466620/cvat.log)
[cvat.db.log](https://github.com/openvinotoolkit/cvat/files/6466621/cvat.db.log)

Additionally, I got the output from bash as below
```bash
> docker logs cvat > cvat.log
> docker logs cvat_db > cvat.db.log
********************************************************************************
WARNING: POSTGRES_HOST_AUTH_METHOD has been set to "trust". This will allow
         anyone with access to the Postgres port to access your database without
         a password, even if POSTGRES_PASSWORD is set. See PostgreSQL
         documentation about "trust":
         https://www.postgresql.org/docs/current/auth-trust.html
         In Docker's default configuration, this is effectively any other
         container on the same system.

         It is not recommended to use POSTGRES_HOST_AUTH_METHOD=trust. Replace
         it with "-e POSTGRES_PASSWORD=password" instead to set a password in
         "docker run".
********************************************************************************
sh: locale: not found
2021-05-12 13:40:17.419 UTC [32] WARNING:  no usable system locales were found

WARNING: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.
2021-05-12 13:40:18.358 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2021-05-12 13:40:18.358 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2021-05-12 13:40:18.358 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2021-05-12 13:40:18.367 UTC [51] LOG:  database system was shut down at 2021-05-12 13:40:18 UTC
2021-05-12 13:40:18.368 UTC [1] LOG:  database system is ready to accept connections
```

**Comment by azhavoro** [comment]
@waterfall414 according to the logs, the database started successfully, same for the cvat processes, but they are in waiting for other services (database, redis, etc.) which cannot be reached. Could you please post the output of `docker network inspect cvat_default` (most likely network name is `cvat_default`, if it's incorrect you can obtain the correct name in `docker network ls` output).

**Comment by azhavoro** [comment]
@waterfall414 the report looks ok and it is identical for my case, could you please do the following experiment with a manually created network:
```
docker run -d --name db1 -e POSTGRES_PASSWORD=password postgres:10-alpine
docker run -d --name db2 -e POSTGRES_PASSWORD=password postgres:10-alpine

docker network create myNetwork
docker network connect myNetwork db1
docker network connect myNetwork db2

# make sure that the network has 2 containers
docker inspect myNetwork

docker exec -it db1 ping db2
```
If that doesn't work either, I assume it is due to some system settings or policies, but the issue is not related to the DNS settings for the docker daemon, because the database IP is successfully resolved by the domainname (i.e. `cvat_db`). Also the problem is not related to proxy settings, ping uses ICMP protocol.

**Comment by waterfall414** [comment]
@azhavoro thanks lot for providing the experiment design and explanation. I've tried it and the ping was successful. 

```bash
> docker run -d --name db1 -e POSTGRES_PASSWORD=password postgres:10-alpine
4388d13d6240f45d54183a9d842c38786535a027a8e80c334b658cb2621a64b0
> docker run -d --name db2 -e POSTGRES_PASSWORD=password postgres:10-alpine
e178745e4791b49ee31a5ba7199aaa05f2bdc31f667eddb28cc924efd4750d7d
> docker network create myNetwork
3c6509ff194bc871e40c29739d0884050c3f4ad6a9c00b15e56eb32e10165170
> docker network connect myNetwork db1
> docker network connect myNetwork db2
> docker inspect myNetwork
[
    {
        "Name": "myNetwork",
        "Id": "3c6509ff194bc871e40c29739d0884050c3f4ad6a9c00b15e56eb32e10165170",
        "Created": "2021-05-13T13:41:44.629289232-04:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": {},
            "Config": [
                {
                    "Subnet": "172.18.0.0/16",
                    "Gateway": "172.18.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "4388d13d6240f45d54183a9d842c38786535a027a8e80c334b658cb2621a64b0": {
                "Name": "db1",
                "EndpointID": "740f8ac009c98f102256287d954628340e082487fe40ed0468d42a7ec18c8765",
                "MacAddress": "02:42:ac:12:00:02",
                "IPv4Address": "172.18.0.2/16",
                "IPv6Address": ""
            },
            "e178745e4791b49ee31a5ba7199aaa05f2bdc31f667eddb28cc924efd4750d7d": {
                "Name": "db2",
                "EndpointID": "4927d64210c4a9279ff68c7b9f2835485d47942baab932d91c97344db8102122",
                "MacAddress": "02:42:ac:12:00:03",
                "IPv4Address": "172.18.0.3/16",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {}
    }
]
> docker exec -it db1 ping db2
PING db2 (172.18.0.3): 56 data bytes
64 bytes from 172.18.0.3: seq=0 ttl=64 time=0.200 ms
64 bytes from 172.18.0.3: seq=1 ttl=64 time=0.093 ms
64 bytes from 172.18.0.3: seq=2 ttl=64 time=0.119 ms
^C
--- db2 ping statistics ---
3 packets transmitted, 3 packets received, 0% packet loss
round-trip min/avg/max = 0.093/0.137/0.200 ms
```
I noticed that the IP of the container starts with "172.1*". That is interesting because the cvat images start with "172.2*", which failed in my case. I made another attempt with nginx, which also gave me the ip 172.17.*. The ping is successful as below.
```bash
> docker run --name some-nginx -v /some/content:/usr/share/nginx/html:ro -d nginx
baf240d3c1aa927509caca66f17d6fb339c8ee7afc3885876ec3f691febb108
> docker inspect some-nginx | grep -i ip
            "IpcMode": "private",
                "HTTP_PROXY=http://amec.zscaler.philips.com:9480",
                "http_proxy=http://amec.zscaler.philips.com:9480",
                "HTTPS_PROXY=http://amec.zscaler.philips.com:9480",
                "https_proxy=http://amec.zscaler.philips.com:9480",
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.4",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
                    "IPAMConfig": null,
                    "IPAddress": "172.17.0.4",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
> ping 172.17.0.4
PING 172.17.0.4 (172.17.0.4) 56(84) bytes of data.
64 bytes from 172.17.0.4: icmp_seq=1 ttl=64 time=0.145 ms
64 bytes from 172.17.0.4: icmp_seq=2 ttl=64 time=0.067 ms
```

Did the experiment narrow down my problem, or actually it becomes more confusing :)

**Comment by waterfall414** [comment]
@azhavoro thanks lot for providing the experiment design and explanation. I've tried it and the ping was successful. 

```bash
> docker run -d --name db1 -e POSTGRES_PASSWORD=password postgres:10-alpine
4388d13d6240f45d54183a9d842c38786535a027a8e80c334b658cb2621a64b0
> docker run -d --name db2 -e POSTGRES_PASSWORD=password postgres:10-alpine
e178745e4791b49ee31a5ba7199aaa05f2bdc31f667eddb28cc924efd4750d7d
> docker network create myNetwork
3c6509ff194bc871e40c29739d0884050c3f4ad6a9c00b15e56eb32e10165170
> docker network connect myNetwork db1
> docker network connect myNetwork db2
> docker inspect myNetwork
[
    {
        "Name": "myNetwork",
        "Id": "3c6509ff194bc871e40c29739d0884050c3f4ad6a9c00b15e56eb32e10165170",
        "Created": "2021-05-13T13:41:44.629289232-04:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": {},
            "Config": [
                {
                    "Subnet": "172.18.0.0/16",
                    "Gateway": "172.18.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "4388d13d6240f45d54183a9d842c38786535a027a8e80c334b658cb2621a64b0": {
                "Name": "db1",
                "EndpointID": "740f8ac009c98f102256287d954628340e082487fe40ed0468d42a7ec18c8765",
                "MacAddress": "02:42:ac:12:00:02",
                "IPv4Address": "172.18.0.2/16",
                "IPv6Address": ""
            },
            "e178745e4791b49ee31a5ba7199aaa05f2bdc31f667eddb28cc924efd4750d7d": {
                "Name": "db2",
                "EndpointID": "4927d64210c4a9279ff68c7b9f2835485d47942baab932d91c97344db8102122",
                "MacAddress": "02:42:ac:12:00:03",
                "IPv4Address": "172.18.0.3/16",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {}
    }
]
> docker exec -it db1 ping db2
PING db2 (172.18.0.3): 56 data bytes
64 bytes from 172.18.0.3: seq=0 ttl=64 time=0.200 ms
64 bytes from 172.18.0.3: seq=1 ttl=64 time=0.093 ms
64 bytes from 172.18.0.3: seq=2 ttl=64 time=0.119 ms
^C
--- db2 ping statistics ---
3 packets transmitted, 3 packets received, 0% packet loss
round-trip min/avg/max = 0.093/0.137/0.200 ms
```
I noticed that the IP of the container starts with "172.1*". That is interesting because the cvat images start with "172.2*", which failed in my case. I made another attempt with nginx, which also gave me the ip 172.17.*. The ping is successful as below.
```bash
> docker run --name some-nginx -v /some/content:/usr/share/nginx/html:ro -d nginx
baf240d3c1aa927509caca66f17d6fb339c8ee7afc3885876ec3f691febb108
> docker inspect some-nginx | grep -i ip
            "IpcMode": "private",
                "HTTP_PROXY=http://aaa:9480",
                "http_proxy=http://aaa:9480",
                "HTTPS_PROXY=http://aaa:9480",
                "https_proxy=http://aaa:9480",
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.4",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
                    "IPAMConfig": null,
                    "IPAddress": "172.17.0.4",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
> ping 172.17.0.4
PING 172.17.0.4 (172.17.0.4) 56(84) bytes of data.
64 bytes from 172.17.0.4: icmp_seq=1 ttl=64 time=0.145 ms
64 bytes from 172.17.0.4: icmp_seq=2 ttl=64 time=0.067 ms
```

Did the experiment narrow down my problem, or actually it becomes more confusing :)

**Comment by azhavoro** [comment]
@waterfall414 Ok, please try to change the network subnet definition to 172.17.0.0/16 (probably need to delete some networks from your experiments before) here https://github.com/openvinotoolkit/cvat/blob/develop/docker-compose.yml#L83 and 172.17.0.1 here https://github.com/openvinotoolkit/cvat/blob/develop/components/serverless/docker-compose.serverless.yml#L17
Hope this helps.

**Comment by waterfall414** [comment]
Cheers! @azhavoro thanks for the guidance. The problem is solved after I changed the ip to 172.18.* instead of 172.17.*

In the beginning, I removed the myNetwork from our previous experiment. But when I first try with the 172.17.*, the docker doesn't allow it, probably because I used it for my nginx experiment.

```bash
# 172.17.0.0/16
> docker-compose up -d
Creating network "cvat_default" with the default driver
ERROR: Pool overlaps with other one on this address space
```

Then I solved the problem by changing 172.17 to 172.18.

Since we get it running, I should close the issue (and surely will). But it is still interesting why 172.28 doesn't work in my case. After I've cloned the cvat repo to my local drive, I tried to run "docker-compose -f docker-compose.yml -f docker-compose.override.yml up -d" with the following override file:

```yml
version: '3.3'

services:
  cvat_proxy:
    environment:
      CVAT_HOST: local_ip_of_the_linux_machine
```

I don't think it messed up the with 172.28 ip route, but by results, it was the only possible cause I can think about now.
